### PR TITLE
[JIT] Allow tensor.device, tensor.dtype, and tensor.shape in JIT

### LINKED ISSUE #12364: [JIT] tensor.device, tensor.dtype, and tensor.shape does not work inside JIT
Sample code:

```python
import torch

@torch.jit.script
def f(x):
    return torch.zeros(5, device=x.device)

x = torch.rand(2,3,4)
print(f(x))
```

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Closes https://github.com/pytorch/pytorch/issues/12364

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
Is the clang-tidy fail related? I'm confused by the output...

**Review by ezyang — APPROVED** [review]
LGTM. (But I'll let someone on JIT merge it.)

File: torch/csrc/jit/ir.h [hunk 1]
```diff
@@ -1092,6 +1092,23 @@ friend struct Block;
...
+      auto* result = create(prim::TensorDevice, {value});
+      result->output()->setType(ListType::create(IntType::get()));
+      return result;
+    } else if (field == "shape") {
+      auto* result = create(prim::TensorShape, {value});
+      result->output()->setType(ListType::create(IntType::get()));
+      return result;
+    }
+    return nullptr;
+  }
```
Comment (apaszke) [inline torch/csrc/jit/ir.h]: I don't think this should live in `ir.h`. It's only used in one single place in the front end, so there's no need to put this in this huge file.

**Comment by zasdfgbnm** [comment]
@apaszke I just deleted `createTensorAttr` and moved everything into `SimpleValue::attr`

**Pull request closed without merging**